src/Postmark/PostmarkClientBase.php

```diff
@@ -140,7 +140,8 @@ protected function processRestRequest($method = NULL, $path = NULL, array $body
 
 		switch ($response->getStatusCode()) {
 			case 200:
-				return json_decode($response->getBody(), true);
+				// Casting BIGINT as STRING instead of the default FLOAT, to avoid loss of precision.
+				return json_decode($response->getBody(), true, 512, JSON_BIGINT_AS_STRING);
```

Comment:
By default, `json_decode` casts `BIGINT` to `float`. This can lead to loss of precision or weird formatting when forming the `URL` with a `float` ID.

`BIGINT` is any integer greater than `PHP_MAX_INT`, a variable that can be 32-bit or 64-bit based on the underlying architecture.

This option casts those numbers as strings. This code gets called by all functions, but I cannot think of any case where we want `BIGINT` to be casted as `float` and lose precision, so I believe this is fully backwards compatible.

Actual floating point numbers are still being casted as `float`.